### PR TITLE
Adds VerticalCardListFeature

### PR DESCRIPTION
**Pull request by vinnyjth**
### What does this PR do, or why is it needed?

Companion PR: https://github.com/ApollosProject/apollos-templates/pull/17

This PR does the following: 

- Adds a VerticalCardListFeature
- Sets up the existing algorithms to be more generic.
- On the template side (see the companion PR) adds the corresponding front end components. 

In general, this gets up to the top of the hill for most of the home feed feature work. 

### What design trade-offs/decisions were made?

Trade offs captured in basecamp.

### How do I test this PR?

1. Pull this PR and it's companion PR
2. Start and link the app. 
3. Visit the home feed, you should see a content card nested between two other features. 

## TODO:

- [x] PR has a relevant title that will be understandable in a public changelog (ie...non developers)
- [ ] Closes [tag-issues-here]
- [x] No new warnings in tests, in storybook, and in-app
- [x] Upload GIF(s) of iOS and Android
- [x] Set a relevant reviewer

## REVIEW:

**Manual QA**

- [ ] Manual QA on iOS and ensure it looks/behaves as expected
- [ ] Manual QA on Android and ensure it looks/behaves as expected

**Code Review: Questions to consider**

- [ ] Read through the "Files changed" tab _very carefully_
- [ ] Edge cases: what assumptions are made about input?
- [ ] What kind of tests could be written?
- [ ] How might we make this easier for someone else to understand?
- [ ] Could the code be simpler?
- [ ] Will the code be easy to modify in the future?
- [ ] What's one part of these changes that makes you excited to merge it?

> The purpose of PR Review is to _improve the quality of the software._

### CODE REVIEW

**Comment by redreceipt** [comment]
Is there a screenshot or a description anywhere? Conceptually I don't really know what I'm reviewing.

**Comment by redreceipt** [comment]
> Is there a screenshot or a description anywhere? Conceptually I don't really know what I'm reviewing.

But of course, this was a relevant question when I was a reviewer :) I'm sure you're still working

File: packages/apollos-data-connector-rock/src/features/data-source.js [hunk 1]
```diff
@@ -200,11 +229,19 @@ Make sure you structure your algorithm entry as \`{ type: 'CONTENT_CHANNEL', aru
...
   }
 
-  getHomeFeedFeatures() {
+  async getHomeFeedFeatures() {
     return Promise.all(
-      get(ApollosConfig, 'HOME_FEATURES', []).map((featureConfig) =>
-        this.createActionListFeature(featureConfig)
-      )
+      get(ApollosConfig, 'HOME_FEATURES', []).map((featureConfig) => {
+        console.log(featureConfig.type);
```
Comment (vinnyjth) [inline packages/apollos-data-connector-rock/src/features/data-source.js]:
```suggestion
```

**Comment by burkeshartsis** [comment]
Alright @vinnyjth I have touched this PR quite a bit and I feel it's in a good spot now. If you don't mind looking over my changes I'm happy to approve this.

**Pull request merged**